src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2586,6 +2586,104 @@ public function get_test_persistent_object_cache() {
 		return $result;
 	}
 
+	/**
+	 * Calculate total amount of autoloaded data.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return int autoloaded data in bytes.
+	 */
+	public function wp_autoloaded_options_size(): int {
+		$alloptions = wp_load_alloptions();
+
+		$total_length = 0;
+
+		foreach ( $alloptions as $option_name => $option_value ) {
+			$total_length += strlen( $option_value );
+		}
+
+		return $total_length;
+	}
+
+	/**
+	 * Callback for autoloaded_options test.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} The test results.
+	 */
+	public function get_test_autoloaded_options() {
+		$autoloaded_options_size  = $this->wp_autoloaded_options_size();
+		$autoloaded_options_count = count( wp_load_alloptions() );
+
+		$base_description = __( 'Autoloaded options are configuration settings for plugins and themes that are automatically loaded with every page load in WordPress. Having too many autoloaded options can slow down your site.' );
+
+		$result = array(
+			'label'       => __( 'Autoloaded options are acceptable' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => __( 'Performance' ),
+				'color' => 'blue',
+			),
+			'description' => sprintf(
+				/* translators: 1. Number of autoloaded options. 2. Autoloaded options size. */
+				'<p>' . esc_html( $base_description ) . ' ' . __( 'Your site has %1$s autoloaded options (size: %2$s) in the options table, which is acceptable.' ) . '</p>',
+				$autoloaded_options_count,
+				size_format( $autoloaded_options_size )
+			),
+			'actions'     => '',
+			'test'        => 'autoloaded_options',
+		);
+
+		/**
+		 * Filters max bytes threshold to trigger warning in Site Health.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param int $limit Autoloaded options threshold size. Default 800000.
+		 */
+		$limit = apply_filters( 'wp_autoloaded_options_limit_size_in_bytes', 800000 );
```

Comment:
Most of the filters in this class use the `site_status_` prefix. These could also be a bit shorter, ex:
```suggestion
		$limit = apply_filters( 'site_status_autoloaded_options_size_limit', 800000 );
```

🔢 Applies throughout.